imports/plugins/core/graphql/server/schemas/cart.graphql

```diff
@@ -0,0 +1,364 @@
+  #token: String
+
+  "TimeStamp at which the entity was last updated."
+  updatedAt: DateTime!
+
+  # Unused until checkout
+  #workflow: Workflow # Needs more consideration
+
+  # Read-only: Carts that don't have an account should be considered anonymous
+  # and should be cleaned up after some time. This should not be set explicitly
+  # on the cart but can be derived and add to the Cart as a read-only
+  # attribute (read, value from a function.) The cart expiration ttl can then be
+  # modified without having to modify cart records.
+  # This value should not be stored in the database.
+  "TimeStamp at which the cart will expire."
+  expiresAt: DateTime!
+}
+
+"Wraps a list of `CartItem`s, providing pagination cursors and information."
+type CartItemConnection implements NodeConnection {
+  edges: [CartItemEdge]
+  nodes: [CartItem]
+  pageInfo: PageInfo!
+  totalCount: Int!
+}
+
+"A connection edge in which each node is a `CartItem` object"
+type CartItemEdge implements NodeEdge {
+  cursor: ConnectionCursor!
+  node: CartItem
+}
+
+"""
+A single item in a cart. The item contains information about an intended purchase.
+"""
+type CartItem implements Node {
+  "A unique identifier for the type."
+  _id: ID!
+
+  "Date first added to cart"
+  addedAt: DateTime! # Should this be "createdAt" ?
+
+  "User defined fields that can be used for any purpose."
+  metafields: [Metafield]
+
+  "Packing information such as item weight, height, length, and depth."
+  parcel: ShippingParcel # Should this be required?
+
+  "The current price of the item"
+  price: Float!
+
+  "The currency for the `price` and `priceWhenAdded` fields"
+  priceCurrency: Currency!
+
+  # This will permit us to perform calculations based on how the price has changed since
+  # the item was added to the cart.
+  "The price at which this item was listed when it was added to the cart"
+  priceWhenAdded: Float!
+
+  # QUESTION: Does productConfiguration satisfy this TODO?
+  # TODO: Variant metadata (from attribute reflection) is needed on the cart item, but should be defined.
+  "The product and chosen options"
+  productConfiguration: ProductConfiguration!
+
+  "The quantity of items that has been added to the cart."
+  quantity: Int!
+
+  # QUESTION: This implies that cart items can be from different shops.
+  # Is it possible that certain shops share a cart but other shops are not
+  # included in that sharing? How is that defined? Probably not applicable on
+  # the cart item but at the cart itself.
+  #
+  # TODO: shopId is not currently required but it should be made required to
+  # avoid issues on cart sharing reconfiguration.
+  "The shop associated with this cart item."
+  shop: Shop
+
+  # TODO: What is this? Simply denormalization? Can this attribute be fetched
+  # directly from the product now in GraphQL?
+  # Should this be embedded within the productConfiguration
+  "The type of product, used to display cart items differently"
+  type: String
```

Comment:
Another flattened field because the product types could be displayed differently and the Cart should be aware. @aldeed a few related comments on this topic so it's clear some Cart concepts should be better communicated in docs. :smile: Sync up with a chat or call tomorrow?